interfaces/sourcegen/sourcegen/_TagFileParser.py

```diff
@@ -0,0 +1,303 @@
+"""Parser for tag files and XML generated by doxygen."""
+
+# This file is part of Cantera. See License.txt in the top-level directory or
+# at https://cantera.org/license.txt for license and copyright information.
+
+import sys
+from pathlib import Path
+import re
```

Comment:
I'd really suggest using a library to parse the xml rather than string operations. Normally I'd recommend lxml but to avoid a third party dependency only for build time it might be better to use `ElementTree` from the stdlib.

Reply:
This is definitely a fair suggestion. This parser is backported from an old branch and works well for the time being, but I'll run a couple of tests before this is ready for review. 

Fwiw, `re` is really fast, given that the tag file itself is already 55k lines long, and there is a myriad of additional XML files that contain details. Not having to parse the entire tree may have some advantages, but there may be a middle ground that's better than the pure `re` approach.

Reply:
Yeah, I'm not so worried about performance, moreso correctness. Although that the file is autogenerated should be a benefit insofar as the tags should all be complete, etc.